gcc/rust/typecheck/rust-hir-type-check-item.h

```diff
@@ -82,7 +82,8 @@ class TypeCheckItem : public TypeCheckBase
 
     context->pop_return_type ();
 
-    expected_ret_tyty->unify (block_expr_ty);
+    if (block_expr_ty->get_kind () != TyTy::NEVER)
+      expected_ret_tyty->unify (block_expr_ty);
```

Comment:
This is not right i think this is where we want to be able to call:

never_type->get_fallback()->unify(block_expr_ty)

Reply:
But it has been checked, in https://github.com/Rust-GCC/gccrs/blob/3c251b99b80c05a7b655aa41b3285b7ce5215ede/gcc/rust/typecheck/rust-hir-type-check-expr.h#L165
Hasn't it?

Reply:
Oh, I noticed if the expression is empty, it won't be checked. https://github.com/Rust-GCC/gccrs/blob/3c251b99b80c05a7b655aa41b3285b7ce5215ede/gcc/rust/typecheck/rust-hir-type-check-expr.h#L150
But we can still fix it in that place?

Reply:
Thats only if its a return expression. What about the case where there is an empty block such as:

```
fn test() -> i32 {
}
```

Here the block will resolve to NeverType in this PR and not be checked against i32 which means we lose the existing check that will sa expected i32 got ().

Reply:
> Oh, I noticed if the expression is empty, it won't be checked.
> 
> https://github.com/Rust-GCC/gccrs/blob/3c251b99b80c05a7b655aa41b3285b7ce5215ede/gcc/rust/typecheck/rust-hir-type-check-expr.h#L148-L152
> 
> But we can still fix it in that place?

I don't think we can generally assume that if the expression resolves to Never it must be checked against the return type.
What about nested if-expressions where there is a missing empty block, since nested if statement need to be dependant on the context of their parent block not nessecarily the return type of a function.

Reply:
> Here the block will resolve to NeverType in this PR and not be checked against i32 which means we lose the existing check that will sa expected i32 got ().
It will resolve to `()`, since the tail is reachable.

Reply:
> What about nested if-expressions where there is a missing empty block, since nested if statement need to be dependant on the context of their parent block not nessecarily the return type of a function.

If there is a missing empty block, it means the tail is reachable so we won't get a NeverType.